lib/twilio/audiohelper.ts

```diff
@@ -832,10 +832,25 @@ class AudioHelper extends EventEmitter {
       //   then that device is unplugged or plugged back in. We can't check for the 'ended'
       //   event or readyState because it is asynchronous and may take upwards of 5 seconds,
       //   in my testing. (rrowland)
-      if (this.inputDevice !== null && this.inputDevice.deviceId === 'default') {
+      const defaultId = 'default';
+      if (
+        // this.inputDevice is not null if audio.setInputDevice() was explicitly called
+        (this.inputDevice && this.inputDevice.deviceId === defaultId) ||
```

Comment:
Nit:

I would prefer to have these complex conditions broken out like so:

```
const defaultId = 'default';
const isInputDeviceSet = this.inputDevice && this.inputDevice.deviceId === defaultId;
const isInputDeviceNotSet = this._defaultInputDeviceStream && this.availableInputDevices.get(defaultId);
```

It might not eliminate the need for the comments entirely, but it helps explain things and make the code more readable IMO.

Reply:
+1. I'll update it. Though I would rename the 2nd one to isDefaultDeviceSet.